pilot/pkg/serviceregistry/kube/controller/ambientindex.go

```diff
@@ -208,15 +209,25 @@ func (c *Controller) Waypoint(scope model.WaypointScope) sets.Set[netip.Addr] {
 	defer a.mu.RUnlock()
 	res := sets.New[netip.Addr]()
 	if addr, f := a.waypoints[scope]; f {
-		if ip, ok := netip.AddrFromSlice(addr.GetIp()); ok {
-			return res.Insert(ip)
+		switch address := addr.Destination.(type) {
+		case *workloadapi.GatewayAddress_Address:
+			if ip, ok := netip.AddrFromSlice(address.Address.GetAddress()); ok {
+				return res.Insert(ip)
+			}
```

Comment:
Same - a small comment on how we would handle waypoints with dual stack or multiple addresses would be nice. 

Also add a comment to make it clear this case will use native CNI do the load balancing ( if I understand the code correctly - we need a lot more comments in the code...).

I would assume this won't work for VMs where there is no CNI ?

Reply:
Also maybe a comment on the entire method for the 'by service account' - there is something hidden in the doc, but would be good to explain it in the code as well. 

When talking to pod - we get the service account of the pod, and lookup a waypoint for same service account. All good.

When talking to a service - not very clear what happens ( neither in the code nor in the doc ).

Reply:
Who is 'we' and 'when' ? Now and with current ztunnel implementation - yes, that's how it works.

Is it a strict requirement that the protocol works this way - and we don't expect it to change in the future, nor other implementations to make different choices ? 

Either way - we need to be clear and explicit in the docs. I don't see a good reason to require this, but as current implementation it is very reasonable.

Reply:
And in particular a non-ambient implementation may connect to waypoints ( which may also be ILBs or external components) in other ways, for example replacing the DNS entry ( no CNI or even ztunnel required ). Or proxyless may implement it differently.